### PR TITLE
Improve links, upload folders, drop target

### PR DESCRIPTION
**Pull request by tenisakb**

### CODE REVIEW

File: src/components/shared/footer.tsx [hunk 1]
```diff
@@ -176,21 +177,27 @@ const Footer = () => (
...
+            target="_blank"
+          >
             Blog <IconArrow src={ICON_ARROW_RIGHT} />
-          </Link>
-          <Link href=" https://www.kucoin.com/trade/OPQ-BTC" target="_blank">
+          </LinkFooter>
+          <LinkFooter
+            href=" https://www.kucoin.com/trade/OPQ-BTC"
+            target="_blank"
+          >
```
Comment (EdmundMai) [inline src/components/shared/footer.tsx]: This doesn't work. As i mentioned in our chat, for external links you’ll need to use `a` elements, not `Link`.

`<a href="google.com" target="_blank">foobar</a>`

`Link` only accepts `to` attribute, which will only link to internal pages while persisting redux state. 
`a` accepts `href`, which can be internal or external. You can add `target="_blank"` to it to open in a new tab.

File: src/components/shared/footer.tsx [hunk 2]
```diff
@@ -176,21 +177,27 @@ const Footer = () => (
...
+          </LinkFooter>
+          <LinkFooter to="/team-page">
             Team <IconArrow src={ICON_ARROW_RIGHT} />
-          </Link>
-          <Link href="https://medium.com/opacity-storage/" target="_blank">
+          </LinkFooter>
+          <LinkFooter
+            href="https://medium.com/opacity-storage/"
+            target="_blank"
+          >
```
Comment (EdmundMai) [inline src/components/shared/footer.tsx]: same for this

File: src/components/shared/footer.tsx [hunk 3]
```diff
@@ -176,21 +177,27 @@ const Footer = () => (
     <Wrapper>
       <Container>
         <FooterLinks>
-          <Link href="https://opacity.io" target="_blank">
+          <LinkFooter href="https://opacity.io" target="_blank">
```
Comment (EdmundMai) [inline src/components/shared/footer.tsx]: and this

File: src/components/shared/footer.tsx (hunk 1 above)
Comment (tenisakb) [reply src/components/shared/footer.tsx]: solved

File: src/components/shared/footer.tsx (hunk 2 above)
Comment (tenisakb) [reply src/components/shared/footer.tsx]: solved

File: src/components/shared/footer.tsx (hunk 3 above)
Comment (tenisakb) [reply src/components/shared/footer.tsx]: solved

File: src/components/file-manager/upload-folder-button.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      onSelected(files);
+    }
+  };
+
+  const addDirectory = node => {
+    if (node) {
+      node.directory = true;
+      node.webkitdirectory = true;
+      node.mozkitdirectory = true;
+    }
```
[the remark this answers is not in the document]
Comment (EdmundMai) [reply src/components/file-manager/upload-folder-button.tsx]: does Opaque even support folder uploads yet?

File: src/components/file-manager/file-manager-slide.tsx [hunk 5]
```diff
@@ -326,6 +328,31 @@ const FolderButton = styled.button`
   }
 `;
 
+const FolderMobileButton = styled.div`
```
[the remark this answers is not in the document]
Comment (tenisakb) [reply src/components/file-manager/file-manager-slide.tsx]: solved

File: src/components/file-manager/upload-folder-button.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    addDirectory(uploadFileInput.current);
+  }, []);
+
+  return (
+    <ThemeProvider theme={theme}>
+      <Fragment>
+        <Button onClick={() => uploadFileInput.current!.click()}>
+          Upload Folder
+        </Button>
+        <form ref={uploadForm} style={{ display: "none" }}>
```
[the remark this answers is not in the document]
Comment (tenisakb) [reply src/components/file-manager/upload-folder-button.tsx]: solved

File: src/components/stands-out/stands-out-slide.tsx [hunk 7]
```diff
@@ -384,22 +386,19 @@ const SubscriptionSlide = ({ history, isLoggedIn }) => (
         </RecommendContentWrapper>
         <PasswordManager>
           <ColumnPasswordManager>
-            <IconPasswordManager
-              src={ICON_LASTPASS}
-              onClick={() => window.open("https://www.lastpass.com/", "_blank")}
-            />
+            <IconLink to="https://www.lastpass.com/" target="_blank">
```
[the remark this answers is not in the document]
Comment (tenisakb) [reply src/components/stands-out/stands-out-slide.tsx]: solved

File: src/components/stands-out/stands-out-slide.tsx [hunk 8]
```diff
@@ -384,22 +386,19 @@ const SubscriptionSlide = ({ history, isLoggedIn }) => (
...
+              <IconPasswordManager src={ICON_LASTPASS} />
+            </IconLink>
           </ColumnPasswordManager>
           <ColumnPasswordManager>
-            <IconPasswordManager
-              src={ICON_1PASSWORD}
-              onClick={() => window.open("https://1password.com/", "_blank")}
-            />
+            <IconLink to="https://www.1password.com/" target="_blank">
```
[the remark this answers is not in the document]
Comment (tenisakb) [reply src/components/stands-out/stands-out-slide.tsx]: solved

File: src/components/stands-out/stands-out-slide.tsx [hunk 9]
```diff
@@ -384,22 +386,19 @@ const SubscriptionSlide = ({ history, isLoggedIn }) => (
...
+              <IconPasswordManager src={ICON_1PASSWORD} />
+            </IconLink>
           </ColumnPasswordManager>
           <ColumnPasswordManager>
-            <IconPasswordManager
-              src={ICON_KEEPASS}
-              onClick={() => window.open("https://keepass.info/", "_blank")}
-            />
+            <IconLink to="https://www.keepass.info/" target="_blank">
```
[the remark this answers is not in the document]
Comment (tenisakb) [reply src/components/stands-out/stands-out-slide.tsx]: solved

File: src/components/file-manager/upload-folder-button.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (tenisakb) [reply src/components/file-manager/upload-folder-button.tsx]: Opaque support create folders and upload files.. in another PR I am planning also create subfolders etc.

File: src/components/file-manager/upload-folder-button.tsx [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+            ref={uploadFileInput}
+            onChange={e => selectFiles()}
+          />
+        </form>
+      </Fragment>
+    </ThemeProvider>
+  );
+};
+
+export default UploadFolderButton;
```
[the remark this answers is not in the document]
Comment (tenisakb) [reply src/components/file-manager/upload-folder-button.tsx]: Ok, fixed

File: src/components/stands-out/stands-out-slide.tsx [hunk 11]
```diff
@@ -311,6 +311,8 @@ const InfoIcon = styled.img`
   }
 `;
 
+const IconLink = styled(Link)``;
```
Comment (EdmundMai) [inline src/components/stands-out/stands-out-slide.tsx]: styled.a

File: src/components/stands-out/stands-out-slide.tsx [hunk 12]
```diff
@@ -414,7 +413,7 @@ const SubscriptionSlide = ({ history, isLoggedIn }) => (
               how our system works.
             </InfoContent>
             <StandsOutButtonLink
-              to="https://github.com/opacity"
+              href="https://github.com/opacity"
               target="_blank"
             >
```
Comment (EdmundMai) [inline src/components/stands-out/stands-out-slide.tsx]: StandsOutButtonLink is a `styled(Link)`. change it to `styled.a`

File: src/components/stands-out/stands-out-slide.tsx (hunk 11 above)
Comment (tenisakb) [reply src/components/stands-out/stands-out-slide.tsx]: fixed

File: src/components/stands-out/stands-out-slide.tsx (hunk 12 above)
Comment (tenisakb) [reply src/components/stands-out/stands-out-slide.tsx]: fixed

**Comment by EdmundMai** [comment]
Did you QA this? It doesn't work for me. It just stalls

Screenshot_8_8_19__11_56_AM

**Comment by EdmundMai** [comment]
1) Looks like folder uploads is broken
Private_file_sharing_and_secure_cloud_storage__powered_by_crypto___Opacity


2) Styles are off:

Private_file_sharing_and_secure_cloud_storage__powered_by_crypto___Opacity

**Comment by EdmundMai** [comment]
Folder uploads still has not been fixed

Screenshot_8_29_19__2_43_PM

**Comment by tenisakb** [comment]
> Folder uploads still has not been fixed
> 
> Screenshot_8_29_19__2_43_PM

its done by the last commit

File: src/components/share-page/share-page-slide.tsx [hunk 13]
```diff
@@ -144,26 +144,29 @@ const SharePageSlide = ({ handle, download }) => {
...
+                  Download file
+                </DownloadButton>
+              </FileInfo>
+            ) : (
+              <Spinner isActive={!error} />
+            )}
+            {error && (
+              <Description>
+                It seems the file you are looking for either does not exist or
+                has been removed. Please check your FIle Handle.
```
Comment (rfornea) [inline src/components/share-page/share-page-slide.tsx]: typo

**Pull request merged**